### DISCUSSION TITLE
How to read package version from file before build

### DISCUSSION THREAD

**Discussion by zyxkad** [python-poetry/poetry#7593] [Q&A] [answered]
I'm trying to build a package use poetry
My file tree is like above:
```
basedir/
 + pluginA/
  |    + pluginfiles ...
 + pyproject.toml
 + plugin.json
```

The `plugin.json` contains the version of my plugin (or the version of my package), so I want to use the version in the `plugin.json`
instead of use the version in `pyproject.toml`. How could I do that?

I think I actually want execute a build script before i build, I saw `build = "build.py"`, but I don't think it's what I want because it won't change the version.

Probably I need create a script to run `poetry build`? I will do that if I don't have any other chance

Thanks in advance

**Comment by PabloEmidio** [comment, marked as answer]
Poetry currently does not support dynamic versioning. It is a work in process in #3332 

But there is https://github.com/mtkennerly/poetry-dynamic-versioning plugin to add this behavior.

**Comment by finswimmer** [comment]
Hey,

could you please explain why you want to have the version in a different location than the `pyproject.toml`?

fin swimmer

**Reply by zyxkad** [reply]
As you see, I'm working on a plugin, the plugin packer use it's own metadata file instead of use the `pyproject.toml`
